src/coreclr/src/jit/codegenarmarch.cpp

```diff
@@ -616,31 +616,32 @@ void CodeGen::genIntrinsic(GenTree* treeNode)
 
     // Right now only Abs/Ceiling/Floor/Round/Sqrt are treated as math intrinsics.
     //
-    switch (treeNode->AsIntrinsic()->gtIntrinsicId)
+    switch (treeNode->AsIntrinsic()->gtIntrinsicName)
     {
-        case CORINFO_INTRINSIC_Abs:
+        case NI_System_Math_Abs:
```

Comment:
Shouldn't most of these have `NI_System_MathF_` cases as well? Where-as CORINFO_INTRINSIC shared the entry, I don't think NamedIntrinics should.

Reply:
I first implemented it with MathF variants, it was quite a noisy change with tons of redundancies (doubles the diff size), as JIT only has a special handling for MathF in case of `_Round` and `_FusedMultiplyAdd`.

Do you see a benefit of separating MathF variants for all methods?

Reply:
The special handling between the double and float variants should always be the saame and they can be purely differentiated by type.

However, it may be confusing to see a named intrinsic called `NI_System_Math_Sqrt` and having to know it also maps to `NI_System_MathF_Sqrt`